jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostGeneralSuite.scala

```diff
@@ -82,19 +82,34 @@ class XGBoostGeneralSuite extends FunSuite with PerTest {
         "objective" -> "binary:logistic").toMap,
       new java.util.HashMap[String, String](),
       numWorkers = 2, round = 5, eval = null, obj = null, useExternalMemory = true,
-      missing = Float.NaN, baseMargin = null)
+      missing = Float.NaN)
     val boosterCount = boosterRDD.count()
     assert(boosterCount === 2)
   }
 
+  test("train with RDD[MLLabeledPoint]") {
```

Comment:
we do not need this test case, what we need is to test different scenarios with the public API

your change is assuming that nothing happens between trainWithRDD and trainDistributed to break the scenarios, e.g. mistakenly swallow some parameters